docs/content/config/advanced/auth-lua.md

```diff
@@ -0,0 +1,194 @@
+local http_header_forwarded_for = "X-Forwarded-For"
+
+package.path = package.path .. ";/etc/dovecot/lua/?.lua"
+local base64 = require("base64")
+
+local http_client = dovecot.http.client {
+    timeout = 1000;
+    max_attempts = 1;
+    debug = false;
+}
+
+function script_init()
+  return 0
+end
+
+function script_deinit()
+end
+
+function is_nextcloud_apppassword(password)
+  return string.find(password, "%w%w%w%w%w%-%w%w%w%w%w%-%w%w%w%w%w%-%w%w%w%w%w%-%w%w%w%w%w") ~= nil
+end
+
+function auth_passdb_lookup(req)
+  if not is_nextcloud_apppassword(req.password)
+  then
+    return dovecot.auth.PASSDB_RESULT_PASSWORD_MISMATCH, ""
+  end
+
+  local auth_request = http_client:request {
+    url = http_url;
+    method = http_method;
+  }
+  auth_request:add_header("Authorization", "Basic " .. base64.encode(req.user .. ":" .. req.password))
+  auth_request:add_header(http_header_forwarded_for, req.remote_ip)
+  local auth_response = auth_request:submit()
+
+  local returnStatus = dovecot.auth.PASSDB_RESULT_INTERNAL_FAILURE
+  local returnDesc = http_method .. " - " .. http_url .. " - " .. auth_response:status() .. " " .. auth_response:reason()
+  if auth_response:status() == http_status_ok
+  then
+    returnStatus = dovecot.auth.PASSDB_RESULT_OK
+    returnDesc = "nopassword=y"
+  elseif auth_response:status() == http_status_failure
+  then
+    returnStatus = dovecot.auth.PASSDB_RESULT_PASSWORD_MISMATCH
+    returnDesc = ""
+  end
+  return returnStatus, returnDesc
+```
+
+Replace the hostname in the URL to the actual hostname of Nextcloud.
+
+Dovecot [provides an HTTP client for use in Lua](https://doc.dovecot.org/admin_manual/lua/#dovecot.http.client). Aside of that, Lua by itself is pretty barebones. It chooses library compactness over included functionality. You can see that in the inefficiently typed regular expression in `is_nextcloud_apppassword()`, which is used because [Lua does not offer full support for regular expressions](https://www.lua.org/pil/20.2.html). You can also see Lua's limited functionality in that a separate library is referenced to add support for Base64 encoding, which is required for [HTTP basic access authentication](https://en.wikipedia.org/wiki/Basic_access_authentication). This library (also a Lua script) is not included. It must be downloaded and stored in the same directory:
+
+```bash
+cd ./docker-data/dms/config/dovecot/lua
+curl -JLO https://raw.githubusercontent.com/iskolbin/lbase64/master/base64.lua
+```
+
+Only use native (pure Lua) libraries as dependencies, such as `base64.lua` from the example. This ensures maximum compatibility. Performance is less of an issue since Lua scripts written for Dovecot probably won't be long or complex, and there won't be a lot of data processing by Lua itself. To see which Lua version is used by Dovecot if you plan to do something that is version dependent, run:
+
+```bash
+docker exec mailserver strings /usr/lib/dovecot/libdovecot-lua.so|grep '^LUA_'
+```
```

Comment:
Shift into Debugging section?

Reply:
Done for the part about finding out which Lua version is used by Dovecot. The part about using additional libraries I want to keep at the place it currently is, since it is essential for Lua development and a bit of a pain to figure out if starting from scratch.

This change is part of the next documentation commit.